src/client/commands/submit.rs

```diff
@@ -2,34 +2,33 @@ use tako::messages::common::ProgramDefinition;
 
 use crate::client::globalsettings::GlobalSettings;
 use crate::client::job::print_job_detail;
-use crate::{rpc_call, Error};
+use crate::{rpc_call};
 use crate::transfer::connection::ClientConnection;
 use crate::transfer::messages::{FromClientMessage, JobStatus, SubmitRequest, ToClientMessage};
+
+use anyhow::anyhow;
 use std::path::PathBuf;
-use crate::common::error::{HqError, invalid_args_error, error};
 
 pub async fn submit_computation(
     gsettings: &GlobalSettings,
     connection: &mut ClientConnection,
     name: Option<String>,
     commands: Vec<String>,
 ) -> crate::Result<()> {
-    let name = match name {
-        None => {
-            commands
-                .get(0)
-                .and_then(|t| {
-                    PathBuf::from(t)
-                        .file_name()
-                        .and_then(|t| t.to_str().map(|s| s.to_string()))
-                })
-                .unwrap_or_else(|| "job".to_string())
-        }
 
-        Some(name) => {
-            validate_and_sanitize_name(name)?
+    let name = name.map(|name| {
```

Comment:
I was also thinking about how to make this part of the code nicer, but I think that it was actually OK before. It is defnitely better to let the user know that there was an error in a proper way, instead of panicking (`unwrap`). So I suggest to return to the previous version with the `match`.